### PR TITLE
Add support for cmath

### LINKED ISSUE #1422: Pyccelisation of a cmath function
### Describe the bug
Pyccel does not support the `cmath` library.

### To Reproduce
Try to pyccelise the following file :
```python
from cmath import cosh
def function(a:'complex'):
    a1 = cosh(a)
    return(a1)
```

### Error details
Provide the generated code, or the error message:
```bash
error [semantic]: cmath_test_function.py [1]| Module cmath is not currently supported by pyccel 
```

### Expected behavior
I suppose that `cmath` library being similar to the `math` one, Pyccel should be able to call it no ?

### Language
Fortran

### PR DESCRIPTION
**Pull request by EmilyBourne**
Add support for the cmath library using similar methods to those used for the math library. Fixes #1422

### CODE REVIEW

File: pyccel/ast/cmathext.py [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+    _shape = None
+    _rank  = 0
+    _order = None
+    def __init__(self, r, phi):
+        super().__init__(r, phi)
+
+#==============================================================================
+# Dictionary to map cmath functions to classes above
+#==============================================================================
+
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/cmathext.py:469]: /bot accept
Code is only provided to ensure the docstrings are aware of the parameter names

File: pyccel/ast/cmathext.py [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+    name = 'polar'
+    _dtype = NativeFloat()
+    _precision = -1
+    _shape = (LiteralInteger(2),)
+    _rank  = 1
+    _order = None
+
+    def __init__(self, z):
+        super().__init__(z)
+
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/cmathext.py:442]: /bot accept
Code is only provided to ensure the docstrings are aware of the parameter names

File: pyccel/ast/cmathext.py [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+    Parameters
+    ----------
+    z : PyccelAstNode
+        The expression passed as argument to the function.
+    """
+    __slots__ = ()
+    name = 'phase'
+    def __init__(self, z):
+        super().__init__(z)
+
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/cmathext.py:417]: /bot accept
Code is only provided to ensure the docstrings are aware of the parameter names

File: ci_tools/bot_messages/bot_commands.txt [hunk 4]
```diff
@@ -5,6 +5,5 @@ This bot reacts to all comments which begin with `/bot`. This phrase can be foll
 - `mark as ready` : Runs the PR tests. If they pass then it adds the appropriate review flag and requests reviews. This command should be used when the PR is first ready for review, or when a review has been answered.
 - `commands` : Shows this list detailing all the commands which are understood.
 - `trust user X` : Tells the bot that a new user X is trusted to run workflows (prevents misuse of GitHub actions for mining etc). This command can only be used by a trusted reviewer.
-- `accept coverage` : Tells the bot that the coverage errors have been investigated and cannot be fixed in this PR.
```
Comment (yguclu) [inline ci_tools/bot_messages/bot_commands.txt:8]: What is this change about?

Comment (EmilyBourne) [reply ci_tools/bot_messages/bot_commands.txt:8]: This command no longer exists. I noticed it was still being described when I wrote the wrong command

File: pyccel/ast/mathext.py [hunk 5]
```diff
@@ -106,91 +106,271 @@ class MathFunctionBool(MathFunctionBase):
...
+    A class which represents a call to the `acosh` function from the `math` library.
+
+    Parameters
+    ----------
+    x : PyccelAstNode
+        The expression passed as argument to the function.
+    """
     __slots__ = ()
     name = 'acosh'
 class MathAsin    (MathFunctionFloat):
```
Comment (yguclu) [inline pyccel/ast/mathext.py:134]: Thanks for adding the docstrings! Given that the code is not as compact as before, it is worth formatting it properly. Here it should be sufficient to add two empty lines between the classes.

Comment (EmilyBourne) [reply pyccel/ast/mathext.py]: Fixed f025067b

File: pyccel/ast/mathext.py [hunk 6]
```diff
@@ -106,185 +109,652 @@ class MathFunctionBool(MathFunctionBase):
...
+    Parameters
+    ----------
+    x : PyccelAstNode
+        The expression passed as argument to the function.
+    """
     __slots__ = ()
     name = 'copysign'
+
+
 class MathCos     (MathFunctionFloat):
```
Comment (yguclu) [inline pyccel/ast/mathext.py:233]: Maybe we should remove these empty spaces as well?

**Review by yguclu — APPROVED** [review]

File: pyccel/codegen/printing/ccode.py [hunk 7]
```diff
@@ -1546,17 +1560,20 @@ def _print_MathFunctionBase(self, expr):
...
-            if arg.dtype != NativeFloat() and not func_name.startswith("pyc"):
-                args.append(self._print(NumpyFloat(arg)))
-            else:
-                args.append(self._print(arg))
+        if expr.dtype is NativeComplex():
+            args = [self._print(a) for a in expr.args]
+        else:
+            args = []
+            for arg in expr.args:
+                if arg.dtype != NativeFloat() and not func_name.startswith("pyc"):
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/codegen/printing/ccode.py]: Fixed 45df9aed

**Pull request merged**